earthkit/data/sources/wekeo.py

```diff
@@ -0,0 +1,161 @@
+# (C) Copyright 2020 ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+#
+
+import os
+
+import hda
+import yaml
+from hda.api import DataOrderRequest
+
+from earthkit.data.core.thread import SoftThreadPool
+from earthkit.data.utils import tqdm
+
+from .file import FileSource
+from .prompt import APIKeyPrompt
+
+
+class HDAAPIKeyPrompt(APIKeyPrompt):
+    register_or_sign_in_url = "https://www.wekeo.eu"
+    retrieve_api_key_url = "https://www.wekeo.eu"
+
+    prompts = [
+        dict(
+            name="url",
+            default="https://wekeo-broker.apps.mercator.dpi.wekeo.eu/databroker",
+            title="API url",
+            validate=r"http.?://.*",
+        ),
+        dict(
+            name="user",
+            example="name",
+            title="User name",
+            hidden=False,
+            validate=r"[0-9a-z]+",
+        ),
+        dict(
+            name="password",
+            example="secretpassword",
+            title="Password",
+            hidden=True,
+            validate=r"[0-9A-z\!\@\#\$\%\&\*]{5,30}",
+        ),
+    ]
+
+    rcfile = "~/.hdarc"
+
+    def save(self, input, file):
+        yaml.dump(input, file, default_flow_style=False)
+
+
+class ApiClient(hda.Client):
+    name = "wekeo"
+
+    def __int__(self, *args, **kwargs):
+        super().__init__(self, *args, **kwargs)
+
+    def retrieve(self, name, request, target=None):
+        matches = self.search(request["request"])
+        out = []
+        for result in matches.results:
+            query = {"jobId": matches.job_id, "uri": result["url"]}
+            url = DataOrderRequest(self).run(query)
+            out.append(
+                os.path.abspath(
+                    self.stream(
+                        result.get("filename"), result.get("size"), target, *url
+                    )
+                )
+            )
+        print(out, ">" * 20)
```

Comment:
This seems to be used only for debugging. Please use LOG.debug() instead print() if this line is really needed.

Reply:
Thanks Sandor I just forgot to remove it.
Now it is removed